packages/cli/src/commands/setup/ui/libraries/chakra-ui.js

```diff
@@ -57,13 +72,39 @@ export async function handler({ force, install }) {
     },
     {
       title: 'Setting up Chakra UI...',
-      skip: () => checkSetupStatus() === 'done',
-      task: () => wrapWithChakraProvider(),
+      skip: () => appJSContains('ChakraProvider'),
+      task: () =>
+        extendAppJS({
+          wrapTag: {
+            wrapperComponent: 'ChakraProvider',
+            wrapperProps: { theme: 'extendedTheme' },
+            wrappedComponent: 'RedwoodApolloProvider',
```

Comment:
I'm afraid this is a bit troublesome. Not all RW apps will have a `<RedwoodApolloProvider>` as we support swapping out the graphql client.

I'm using ReactQuery in one of my apps where my `App.js` `App` component looks like this:
```
const App = () => (
  <ThemeProvider theme={theme}>
    <FatalErrorBoundary page={FatalErrorPage}>
      <RedwoodProvider>
        <QueryClientProvider client={queryClient}>
          <RedwoodReactQueryProvider>
            <SettingsProvider>
              <CartProvider>
                <SearchProvider>
                  <Routes />
                </SearchProvider>
              </CartProvider>
            </SettingsProvider>
          </RedwoodReactQueryProvider>
        </QueryClientProvider>
      </RedwoodProvider>
    </FatalErrorBoundary>
  </ThemeProvider>
)
```

Reply:
You're right, that is troublesome. Note that this problem exists currently in mainline - I suspect if you ran `yarn rw setup ui chakra-ui` for the above app, it wouldn't work as expected. The existing code also hard-codes `RedwoodApolloProvider`. That said, that's no excuse to leave it broken!

So we need some way to ask a Redwood app, "what's your graphql provider?" I don't know how to do that, or if it's possible. Do you know a way?

Reply:
We are pretty comfortable offering "golden path" support if you stick to the Redwood defaults. Otherwise, we just need to communicate well but there are no expectations for us to handle customized (and edge) cases.

For this situation, it might mean something like the following:
1. use regex to check for `<RedwoodApolloProvider>`
2. if not found, then output clear messaging about a) cannot find X and b) manual step required c) instructions how to complete the manual step
3. do not fail setup; finish and output a clear "You still need to complete the setup" message

What do you think?

Reply:
That sounds reasonable to me, but I think we can go a step further to help edge-case users. We could add another argument to the setup command, say, `--graphql-provider`, and default it to `RedwoodApolloProvider`. Then, if it isn't found, we error out with a very direct call to action: 

> It looks like you're not using RedwoodApolloProvider as your graphql provider. Please rerun setup with `--graphql-provider` set, e.g. `--graphql-provider RedwoodReactQueryProvider`.

Reply:
On second thought, that flag name is misleading. Unless there's some reason the ui component provider must directly wrap the GraphQL provider, 
we don't have to mention graphql at all to the end user. 

We have the ability to wrap the component provider around any tag, so we might as well name the flag `wrapped-component` (or something) and then just wrap the component provider around whatever is given. 

We can still default to `ApolloRedwoodProvider`, and then our flag name conveys the actual flexibility of the choice.

Reply:
If you look at my `App.js` above you'll notice another component in the tree, `<RedwoodProvider>`. That one is always going to be there. So how about you look for that instead, and put the Chakra provider just inside that one, instead of just outside `<ApolloRedwoodProvider>`?

Reply:
That could work too. I’m realizing I don’t understand the implications of shuffling around this tag. Is there any meaningful difference between putting these provider classes in one order vs another? If not, I’d support your approach which just hard codes any tag reasonably guaranteed to be there.

Reply:
With the default App.tsx
```
const App = () => (
  <FatalErrorBoundary page={FatalErrorPage}>
    <RedwoodProvider titleTemplate="%PageTitle | %AppTitle">
      <RedwoodApolloProvider>
        <Routes />
      </RedwoodApolloProvider>
    </RedwoodProvider>
  </FatalErrorBoundary>
)
```
"Just inside `<RedwoodProvider>`" and "Wrapped around `<RedwoodApolloProvider>`" should be the same thing - no shuffling around of tags - the order stays the same. The only difference is what tag you use to anchor your logic to. And yes, I'd say `<RedwoodProvider>` is reasonably guaranteed to be there.

Reply:
I see. This is simpler than I thought at first. Thanks for explaining - I'll change the implementation accordingly.

Reply:
This has been resolved. I've supplemented the `extendAppJS` function with the following:
```javascript
export function extendAppJS({
  insertComponent: {
    name = undefined,
    props = undefined,
    around = undefined,
    within = undefined,
    insertBefore = undefined,
    insertAfter = undefined,
  },
  imports = [],
  moduleScopeLines = [],
}) {
```

The caller can now specify if they want their inserted component put `within` a target component, or `around` a target component. I've tested both behaviors pretty thoroughly, and have updated chakra and mantine to now insert their providers `within` `RedwoodProvider`, as described above.

Reply:
@Tobbe if you're satisfied, please resolve this conversation. Thank you for your thoughtful review here - we now have a better API and covered way more corner cases!